src/os/_Private.php

```diff
@@ -20,3 +20,18 @@ function errno(): ?OS\Errno {
   $errno = \posix_get_last_error() as int;
   return $errno === 0 ? null : OS\Errno::assert($errno);
 }
+
+trait ExceptionWithErrnoTrait<T as ?OS\Errno> {
+  require extends \Exception;
+
+  private T $errno;
+
+  final public function __construct(T $errno) {
+    $this->errno = $errno;
+    parent::__construct();
```

Comment:
Not related to this PR, but do we not want these to have additional arguments that would be passed to the standard Exception constructor (message, wrapped exception, ...)?

Reply:
Yep; I think that needs addressing when @azjezz's overall point about specificity of exception classes is addressed - especially RE whether exception **classes** should indicate which operation failed, why the operation failed, both, or some blend.

Will follow up on #37